podbuilder/podbuilder.go

```diff
@@ -310,6 +311,14 @@ func (pb *PodBuilder) newPod() *corev1.Pod {
 		})
 	}
 
+	pb.run.Env = append(pb.run.Env, corev1.EnvVar{
+		Name:  "POD_TIMEOUT",
+		Value: strconv.FormatInt(int64(pb.test.Spec.TimeoutSeconds), 10)})
+
+	pb.run.Env = append(pb.run.Env, corev1.EnvVar{
+		Name:  config.KillAfter,
+		Value: pb.defaults.KillAfter})
+
```

Comment:
I add the environment variables here since we would also want to have similar features for workers.